cachi2/core/package_managers/npm.py

```diff
@@ -177,7 +194,10 @@ def get_package_name_from_path(package_path: str) -> str:
         for package_path, package_data in self._lockfile_data.get("packages", {}).items():
             # ignore links and the main package, since they're already accounted
             # for elsewhere in the lockfile
-            if package_data.get("link") or package_path == "":
+            if package_data.get("link"):
+                self._workspaces.append(package_data.get("resolved"))
+                continue
```

Comment:
This would find not only workspaces, but also non-workspace `file:path/to/dir` dependencies

If this is supposed to be just workspaces, we should only take the paths that match the glob patterns in `.packages[""].workspaces` (glob matching - https://docs.python.org/3/library/fnmatch.html)

Reply:
I updated this part.